### ISSUE TITLE
file question

### ISSUE THREAD

**Issue by linlinx666** [muzishen/IMAGDressing#54]
I have currently trained a. pt model file through your code. If I want to proceed with the evaluation process, in short, what else do I need to do? The required file cloth_sth, Can I understand the following: The only function of my trained model in the evaluation process is to turn the cloth into a cloth_mask, and can I use IGpair's clothes for the cloth_path for evaluation, the IGpair's open pose file as the required pose file for evaluation, and the IGpair's JSON file as the text prompts? Face-PATH is the face file you provided yesterday. If possible, what should be the model-PATH file, what should be inside it, where can I find this content, and where can I find it? Thank you

**Comment by WhiteJiang** [comment]
@linlinx666 After training the model, you should use the trained model and reference clothes to generate model images, extract clothing masks based on the generated model images, and evaluate them with the reference clothes. model_path should be the model images you generated using the model.

**Comment by linlinx666** [comment]
1.Could U please tell me use which file to generate model images，and use which file to transform generated model images into cloth_mask 
2.can i use IGpairs cloth as cloth_path file 
    can i use IGpairs openpose as pose_path file
    can i use IGpairs face as face_path file
thank u sincerely

**Comment by WhiteJiang** [comment]
@linlinx666 1. You can use [inference_IMAGdressing.py (https://github.com/muzishen/IMAGDressing/blob/main/inference_IMAGdressing.py) to generate model images , and the model from the https://github.com/GoGoDuck912/Self-Correction-Human-Parsing/tree/master to transform generated model images into cloth_mask
2.  Of course you can, but please make sure you don't use this data for training

**Comment by linlinx666** [comment]
Why can't I use this data for training and what consequences will it lead to

**Comment by WhiteJiang** [comment]
@linlinx666 Based on the principle of fair evaluation, the training data used by the model is not allowed to be used for evaluation.

**Comment by linlinx666** [comment]
So if I use your code to train on the IGPAIR dataset, then for the evaluation process, I would need to use cloth from a different dataset. I would generate the model image using the files you mentioned, then generate the cloth_mask from the model image, and I should not use the face and openpose files you provided. Instead, I should use the corresponding files from a different dataset. Is this correct?

**Comment by WhiteJiang** [comment]
Yes, we also provide IGPair_Test for evaluation. If you do not use openpose and face files for training, you can also use them for evaluation.

**Comment by linlinx666** [comment]
I did not modify the IGPair folder during training, and the original IGPair folder did not contain the face files. Does this mean that I did not use face data during the training process? I'm also unsure if I used openpose for training. In this case, can I still use eval_s.py for evaluation instead of eval.py?

**Comment by WhiteJiang** [comment]
Yes, the original training code does not use face and openpose files

**Comment by linlinx666** [comment]
So, should I not use the openpose and face files for evaluation in my process? However, eval_s.py requires these files. Does this mean I can only use eval.py for evaluation?

**Comment by WhiteJiang** [comment]
You can use eval_s.py with the openpose and face files

**Comment by linlinx666** [comment]
Now, assuming that I did not use openpose during training (although there is an openpose file in IGPair) and did not use the face file (since there is no face file in IGPair), can I use the face file you provided and the openpose file from IGPair for evaluation

**Comment by linlinx666** [comment]
Here’s the translation of your question:

"I have another question. If the cloth_path file I want to use for evaluation is from a different dataset's cloth, can I still use the IGPair.json file from IGPair for the required text_prompts during evaluation?"

**Comment by WhiteJiang** [comment]
1. yes
2. The text prompt in IGPair.json is generated for the corresponding model image and is not recommended.
3. You can use a simple description, such as "A smiling lady" as your text prompt. You can also generate a corresponding prompt for cloth.

**Comment by linlinx666** [comment]
You mentioned generating text prompts. How should I generate text prompts, and which file or method should I use to generate them?

Thank you for your patience!

**Comment by WhiteJiang** [comment]
You can design the text prompt as you like.  For example，"A young woman" "An old woman sitting on a park bench" "A cute girl with big eyes" "A happy girl on the beach"

**Comment by linlinx666** [comment]
Do you mean that the text_prompts need to be manually written one by one with descriptions, rather than being automatically generated?

**Comment by WhiteJiang** [comment]
Writing manually is one way, you can also use other automated tools, such as LLM. If you have other questions, adding WeChat friends may be a better choice. My email is xinjiang@njust.edu.cn
